blocks/embed/embed.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright 2021 Adobe. All rights reserved.
+ * This file is licensed to you under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License. You may obtain a copy
+ * of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under
+ * the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR REPRESENTATIONS
+ * OF ANY KIND, either express or implied. See the License for the specific language
+ * governing permissions and limitations under the License.
+ */
+/* global window */
+/* eslint-disable import/named, import/extensions */
+
+import { buildFigure } from '../../scripts/scripts.js';
+
+const loadScript = (url, callback, type) => {
+    const $head = document.querySelector('head');
+    const $script = document.createElement('script');
+    $script.src = url;
+    if (type) {
+      $script.setAttribute('type', type);
+    }
+    $head.append($script);
+    $script.onload = callback;
+    return $script;
+}
+
+// 'open.spotify.com' returns 'spotify'
+const getServer = (url) => {
```

Comment:
i am having a bit of a hard time deciphering what this does. the comment was definitely helpful. but i think this is equivalent to:

```
url.hostname.split('.')[1]
```

which i am not sure deserves its own function.

Reply:
Also, it doesn't look flexible enough to deal with various different hostname formats, e.g. `youtu.be`. I would probably make this part of the `EMBEDS_CONFIG ` map instead.